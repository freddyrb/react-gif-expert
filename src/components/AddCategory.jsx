import { useState } from "react";
import PropTypes from 'prop-types'

export const AddCategory = ({ onNewCategory }) => {

    const [inputValue, setInputValue] = useState('');

    const onInputChange = ({ target }) => {
        
        setInputValue(target.value);
    }

    const onSubmitForm = (event) => {

        event.preventDefault();
        console.log(inputValue);

        if(inputValue.length <= 1) return;

        onNewCategory(inputValue.trim());

        //setCategories( categoriess => [ inputValue,...categoriess]);
        setInputValue('');

    }

    return (
        <form aria-label="form" onSubmit={  onSubmitForm }>
            <input
                type="text"
                placeholder="Search by"
                value={inputValue}
                onChange={onInputChange}></input>
        </form>
    );
};


AddCategory.propTypes = {

    onNewCategory: PropTypes.func.isRequired

}