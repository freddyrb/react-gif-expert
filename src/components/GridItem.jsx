
import PropTypes from 'prop-types';

export const GridItem = ({ title, url }) => {


  return (
    <div className="card">
      <img src={url} alt={title}></img>
      <p>{title}</p>
    </div>
  )
}


GridItem.propTypes = {

  title: PropTypes.string.isRequired ,
  url: PropTypes.string.isRequired

}

/*

Task:

1. Add PropTypes ?????? yarn add
a. title required
b. url required

2. Eval the snapshot

*/