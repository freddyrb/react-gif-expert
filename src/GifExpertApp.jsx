import { useState } from 'react';
import { AddCategory , GifGrid} from './components';


const apiKey = 'TDMEPpzOGNU7LffBaxHcoUgjJioYznUX';

export const GifExpertApp = () => {

    const [categories, setCategories] = useState(['Developers']);

    console.log(categories);

    const onAddCategory = (newCategory) => {

        //categories.push(newCategory);

        if (categories.includes(newCategory)) return;
        setCategories([ newCategory, ...categories]);
        console.log('Father function called from child')

    }

    return (
        <>
            {/* titulo */}
            <h1>GifExpertApp</h1>

            {/* input */}
            <AddCategory
                //setCategories = { setCategories }
                onNewCategory={event => onAddCategory(event)}
            />

            {/* listado de gifs */}

            {
                categories.map(category => (

                    <GifGrid key={category} category={category} />

                ))
            }

        </>
    );
};