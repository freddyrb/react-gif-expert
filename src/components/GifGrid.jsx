import { GridItem } from "./GridItem.jsx";
import { useFetchGfis } from "../hooks/useFetchGifs.js";
import PropTypes from 'prop-types'


export const GifGrid = ({ category }) => {

    const { images, isLoading } = useFetchGfis(category);

    console.log(images, isLoading);

    return (
        <>

            <h3>{category}</h3>
            {isLoading && (<h2>Loading...</h2>)}
            <div className="card-grid">
                {
                    images.map((image) => (
                        <GridItem
                            key={image.id}
                            {...image}
                        />
                    ))
                }
            </div>



        </>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}
