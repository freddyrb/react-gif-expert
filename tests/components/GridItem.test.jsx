import { queryByTitle, render, screen } from "@testing-library/react"
import { GridItem } from "../../src/components"

describe('Test on GridItem', () => { 
    
    const title = 'Hi';
    const url = 'https://www.oshyn.com/';
    

    test('should match the snapshot', () => { 
        
        const {container} = render(<GridItem title={title } url={ url }/>);

        expect(container).toMatchSnapshot();

     });

     test('should show the image with the url and Alt given', () => { 
        
        render(<GridItem title={title } url={ url }/>);

        console.log(url)
        console.log(screen.getByRole('img').src);
        //expect( screen.getByRole('img').alt).toBe(title);

        const { src, alt } = screen.getByRole('img');

        expect(src).toBe(url);
        expect(alt).toBe(title);


      });

      test('should show the component title', () => { 
        
        render(<GridItem title={title } url={ url }/>);

        expect(screen.getByText(title)).toBeTruthy();

       })

 })