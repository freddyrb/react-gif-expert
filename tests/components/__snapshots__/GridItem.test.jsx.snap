// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on GridItem should match the snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Hi"
      src="https://www.oshyn.com/"
    />
    <p>
      Hi
    </p>
  </div>
</div>
`;
